### PR TITLE
Update CI jobs config for ironic-agent-image in OCP 4.17

### PR DESCRIPTION
**Pull request by MahnoorAsghar**

### CODE REVIEW

**Comment by MahnoorAsghar** [comment]
/hold cancel
OCP 4.17 branching done today

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-ovn-dualstack

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-scos-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-prevalidation-e2e-metal-ipi-serial-ipv4-prevalidation

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-prevalidation-e2e-metal-ipi-ovn-ipv6-prevalidation

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-prevalidation-e2e-metal-ipi-ovn-dualstack-prevalidation

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-prevalidation-e2e-metal-ipi-ovn-ipv6-prevalidation

**Comment by iurygregory** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-scos-images

**Comment by iurygregory** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-images

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16.yaml [hunk 1]
```diff
@@ -15,8 +15,7 @@ images:
   to: ironic-agent
 promotion:
   to:
-  - disabled: true
```
Comment (iurygregory) [inline ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16.yaml:18]: I'm ok with the change, but looking at how we did back to move to 4.16 https://github.com/openshift/release/pull/46676
We don't update the `openshift-ironic-agent-image-release-4.<previous release>yaml` 
Is this change really necessary? Also I'm a bit puzzled with the disabled we had before so maybe is ok...

File: ci-operator/jobs/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16-presubmits.yaml [hunk 2]
```diff
@@ -475,6 +475,7 @@ presubmits:
         - --image-import-pull-secret=/etc/pull-secret/.dockerconfigjson
         - --report-credentials-file=/etc/report/credentials
         - --target=[images]
+        - --target=[release:latest]
```
Comment (iurygregory) [inline ci-operator/jobs/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16-presubmits.yaml:478]: Do we really need this? I also couldn't find a similar change when we moved to 4.16

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-scos-images

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16.yaml (hunk 1 above)
Comment (MahnoorAsghar) [reply ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16.yaml:18]: There was a failing CI job without this IIRC. From what I understand promotion for only the latest release should be disabled, and it should be enabled for everything else. For the PR #46676, the promotion was already disabled for OCP 4.15 ([link](https://github.com/elfosardo/release/blob/1d81405463d77b9e40d3a5f2621382795380bf54/ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.15.yaml#L17)).

Comment (MahnoorAsghar) [reply ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-release-4.16.yaml:18]: (I basically went to he origin branch of the PR, made sure the last commit hash is the same as the PR, and found the file for 4.15.)

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-prevalidation-e2e-metal-ipi-serial-ipv4-prevalidation

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-e2e-metal-ipi-bm

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-ovn-dualstack

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-okd-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-e2e-metal-ipi-bm

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-ovn-dualstack

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-bm

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-ovn-ipv6

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-serial-ipv4

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-bm

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-prevalidation-e2e-metal-ipi-ovn-dualstack-prevalidation

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-e2e-metal-ipi-serial-ipv4

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-virtualmedia

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-e2e-metal-ipi-ovn-dualstack

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-e2e-metal-ipi-serial-ipv4

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-release-4.17-okd-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-scos-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-scos-images

**Comment by MahnoorAsghar** [comment]
/pj-rehearse pull-ci-openshift-ironic-agent-image-main-okd-scos-images
